grails-app/views/report/showTransactionReport.gsp

```diff
@@ -259,7 +259,14 @@
 				data.push({ name: "startDate", value: $("#startDate").val() });
 				data.push({ name: "endDate", value: $("#endDate").val() });
 				data.push({ name: "category", value: $("#category").val() });
-                data.push({ name: "products", value: $("#products-select").val() });
+
+                let selectedProducts = $("#products-select").val();
+                if (selectedProducts) {
+                  selectedProducts.forEach(productId => {
+                    data.push({ name: "products", value: productId });
+                  });
+                }
```

Comment:
This is what makes it produce `products=1&products=2` which is the preferred approach